DQM/Physics/src/SMPDQM.cc

```diff
@@ -0,0 +1,299 @@
+
+  Nelecs = bei.book1D("Nelecs", "Number of electrons", 20, 0., 10.);
+  HoverE_elecs = bei.book1D("HoverE_elecs", "HoverE", 50, 0., 1.);
+  pt_elecs = bei.book1D("pt_elecs", "p_{T}(elecs)", 40, 0., 200.);
+  eta_elecs = bei.book1D("eta_elecs", "#eta(elecs)", 50, -5., 5.);
+  phi_elecs = bei.book1D("phi_elecs", "#phielecs)", 32, -3.2, 3.2);
+  elIso_cal = bei.book1D("elIso_cal", "Iso_{cal}^{el}", 21, -1., 20.);
+  elIso_trk = bei.book1D("elIso_trk", "Iso_{trk}^{el}", 21, -2., 40.);
+  elIso_CombRelIso = bei.book1D("elIso_CombRelIso", "Iso_{rel}^{el}", 20, 0., 1.);
+
+  PFJetpt = bei.book1D("PFJetpt", "p_{T}(jets)", 100, 0.0, 100);
+  PFJeteta = bei.book1D("PFJeteta", "#eta(jets)", 50, -2.5, 2.5);
+  PFJetphi = bei.book1D("PFJetphi", "#phi(jets)", 50, -3.14, 3.14);
+  PFJetMulti = bei.book1D("PFJetMulti", "jet multiplicity", 5, -0.5, 4.5);
+  PFJetRapidity = bei.book1D("PFJetRapidity", "y(jets)", 50, -6.0, 6.0);
+  mjj = bei.book1D("mjj", "m_{jj}", 100, 0, 1000);
+  detajj = bei.book1D("detajj", "#Delta#etajj", 20, 0, 5);
+
+  dphi_lepMET = bei.book1D("dphi_lepMET", "#Delta#phi(lep,MET)", 60, -3.2, 3.2);
+  mass_lepMET = bei.book1D("mass_lepMET", "m(lep,MET)", 200, 0, 200);
+  pt_lepMET = bei.book1D("pt_lepMET", "p_{T}(lep,MET)", 200, 0, 200);
+  detall = bei.book1D("detall", "#Delta#etall", 20, -5, 5);
+  dphill = bei.book1D("dphill", "#Delta#phill", 20, -6.4, 6.4);
+  mll = bei.book1D("mll", "mll", 200, 0, 200);
+  etall = bei.book1D("etall", "#Delta#etall", 60, -6, 6);
+  ptll = bei.book1D("ptll", "p_{T}ll", 200, 0, 200);
+  mjj = bei.book1D("mjj", "mjj", 100, 0, 1000);
+  detajj = bei.book1D("detajj", "#Delta#etajj", 20, 0, 5);
+
+  dphi_lepjet1 = bei.book1D("dphi_lepjet1", "#Delta#phi(lep,jet1)", 60, -3.2, 3.2);
+
+  dphi_lep1jet1 = bei.book1D("dphi_lep1jet1", "#Delta#phi(lep1,jet1)", 60, -3.2, 3.2);
+  dphi_lep2jet1 = bei.book1D("dphi_lep2jet1", "#Delta#phi(lep2,jet1)", 60, -3.2, 3.2);
+}
+void SMPDQM::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+  std::vector<TLorentzVector> recoPFJets;
+  recoPFJets.clear();
+  TLorentzVector imet;
+  imet.Clear();
+  std::vector<TLorentzVector> selected_recoPFJets;
+  selected_recoPFJets.clear();
+  std::vector<TLorentzVector> selected_lep;
+  selected_lep.clear();
+
+  bool verbose = false;  //true;
+
+  for (std::vector<edm::EDGetTokenT<edm::View<reco::MET>>>::const_iterator met_ = mets_.begin(); met_ != mets_.end();
+       ++met_) {
+    edm::Handle<edm::View<reco::MET>> met;
+    if (!iEvent.getByToken(*met_, met))
+      continue;
+    if (met->begin() != met->end()) {
+      MET->Fill(met->begin()->et());
+      METphi->Fill(met->begin()->phi());
+      imet.SetPtEtaPhiM(met->begin()->et(), 0., met->begin()->phi(), 0.0);
+    }
+  }
+
+  // Muons
+
+  edm::Handle<edm::View<reco::Vertex>> pvs;
+  if (!iEvent.getByToken(pvs_, pvs)) {
+    return;
+  }
+
+  unsigned int pvMult = 0;
+
+  if (verbose)
+    cout << "check this" << pvs->size() << endl;
```

Comment:
@anmehta general comment here and elsewhere: please replace ```cout``` with a suitable choice for the MessageLogger, depending on the severity of the message. ```cout``` is not thread safe